### ISSUE TITLE
Change Request: Improve error location for `no-sparse-arrays`

### ISSUE THREAD

**Issue by fisker** [eslint/eslint#18569]
### ESLint version

9.4.0

### What problem do you want to solve?

https://eslint.org/play/#eyJ0ZXh0IjoiW1xuYSxcbmIsXG5jLFxuZCwsXG5lLFxuZiwgXG5dIiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby1zcGFyc2UtYXJyYXlzIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

The error range is too wide, expect explicit location for holes.

### What do you think is the correct solution?

Point to the comma instead of the whole array.

### Participation

- [x] I am willing to submit a pull request for this change.

**Comment by fasttime** [comment]
Thanks for the issue @fisker. To be clear, are you suggesting that the rule should report a problem for each single gap instead of once for a whole sparse array? For example:

```
/* eslint no-sparse-arrays: "error" */

const a = [, "foo", , , "bar", "baz", ,];
```

Should the rule report four errors for the above code?

**Comment by fisker** [comment]
Yes, 4 errors. In big array, it's very hard to find the hole.

**Comment by fisker** [comment]
The error message also says `Unexpected comma in middle of array. ` so it should make sense to point to the comma.

**Comment by fasttime** [comment]
So this request is about changing the default behavior of the `no-sparse-arrays` rule. Rather than reporting on a whole sparse array, the rule shall report on each gap in the element list of a sparse array, with the report location on the comma that follows the gap.

![image](https://github.com/eslint/eslint/assets/6367844/ad09dec4-1b6e-4dd8-9c8f-f180d8524796)

This should make it easier to locate unexpected commas in a large array and it should align the rule behavior with the message that says "Unexpected comma in middle of array."

I couldn't find a similar request in our previous issues, so I'd like to hear what the team thinks about this. @eslint/eslint-team

**Comment by JoshuaKGoldberg** [comment]
+1, I'm in support of this. It's confusing as a user to have error report ranges include parts of code that aren't actually problems. 

Anecdotally speaking, most of the time I've seen this rule report have been accidental double-commas. If the developer didn't realize they'd added an extra comma then having the rule show the extra comma -rather than give a vague _"it's somewhere in here"_- would be quite nice.

**Comment by nzakas** [comment]
@fisker please go ahead and submit a PR for the change.
